Resources/Prototypes/Entities/Objects/Weapons/Guns/flare_gun.yml

```diff
@@ -1,39 +1,96 @@
 - type: entity
   name: flare gun
-  parent: BaseWeaponLauncher
+  parent: BaseItem
   id: WeaponFlareGun
   description: A compact, single-shot pistol that fires shotgun shells.
   components:
   - type: Sprite
     sprite: Objects/Weapons/Guns/Shotguns/flaregun.rsi
     layers:
-      - state: icon
+      - state: base
         map: ["enum.GunVisualLayers.Base"]
   - type: Item
     size: Small
     sprite: Objects/Weapons/Guns/Shotguns/flaregun.rsi
+  - type: ItemSlots
+    slots:
+      gun_chamber:
+        name: Chamber
+        startingItem: ShellShotgunFlare
+        priority: 1
+        whitelist:
+          tags: ## TODO: Add a risk of the gun blowing up if using non-light shotgun shells, and then re-enable them.
+            ## - ShellShotgun
+            - ShellShotgunLight
+  - type: ContainerContainer
+    containers:
+      gun_chamber: !type:ContainerSlot
+  - type: ChamberMagazineAmmoProvider
+    autoCycle: false
+    boltClosed: true
+    soundBoltClosed: /Audio/Weapons/Guns/Cock/revolver_cock.ogg
+    soundBoltOpened: /Audio/Weapons/Guns/Cock/revolver_cock.ogg
+    soundRack: /Audio/Weapons/Guns/Cock/revolver_cock.ogg
+  - type: Clothing
+    sprite: Objects/Weapons/Guns/Shotguns/flaregun.rsi
+    quickEquip: false
+    slots:
+    - Belt
+    - suitStorage
+  - type: Appearance
   - type: Gun
     fireRate: 8
     selectedMode: SemiAuto
     availableModes:
       - SemiAuto
     soundGunshot:
       path: /Audio/Weapons/Guns/Gunshots/flaregun.ogg
-  - type: BallisticAmmoProvider
-    whitelist:
-      tags:
-        - ShellShotgun
-    proto: ShellShotgunFlare
-    capacity: 1
-    soundInsert:
-      path: /Audio/Weapons/Guns/MagIn/shotgun_insert.ogg
+
+
+- type: entity
+  name: security shell gun
+  parent: [BaseItem, BaseSecurityContraband]
```

Comment:
The shell gun should be inheriting from WeaponFlareGun. There are a lot of duplicate fields between the two prototypes